analytics/src/main/java/com/segment/analytics/internal/Utils.java

```diff
@@ -235,6 +235,7 @@ public static String assertNotNullOrEmpty(String text, @Nullable String name) {
   /** Returns an immutable copy of the provided map. */
   @NonNull
   public static <K, V> Map<K, V> immutableCopyOf(@NonNull Map<K, V> map) {
+    // TODO (major version change) change this out to allow @Nullable input and guard against it
```

Comment:
ideally this would become 
```
@NonNull
  public static <K, V> Map<K, V> immutableCopyOf(@Nullable Map<K, V> map) {
    if (isNullOrEmpty(map)) {
      return Collections.emptyMap();
    }
    return Collections.unmodifiableMap(new LinkedHashMap<>(map));
  }
```
but dont wanna do a breaking change in this version.

Reply:
👍

Reply:
Might be good to leave that in the TODO comment in the code itself as well.